# Секреты темнолесья. История видеоигр.
<br/>
<div class="alert alert-info">

- Автор: Данилевич Елена

- Дата: 2.02.2026
</div>

### Цели и задачи проекта

<br/>
<div class="alert alert-info">
Цель проекта: Изучить развитие игровой индустрии с 2000 по 2013 год

Задачи проекта:

* изучить данные, проверить их корректность, провести предобработку и получить необходимый срез данных
* сделать обзор игровых платформ
* изучить объемы продаж игр разных жанров
* изучить региональные предпочтения игроков
</div>


### Описание данных

<br/>
<div class="alert alert-info">
Данные `/datasets/new_games.csv` содержат информацию о продажах игр разных жанров и платформ, а также пользовательские и экспертные оценки игр:

`Name` — название игры.

`Platform` — название платформы.

`Year of Release` — год выпуска игры.

`Genre` — жанр игры.

`NA sales` — продажи в Северной Америке (в миллионах проданных копий).

`EU sales` — продажи в Европе (в миллионах проданных копий).

`JP sales` — продажи в Японии (в миллионах проданных копий).

`Other sales` — продажи в других странах (в миллионах проданных копий).

`Critic Score` — оценка критиков (от 0 до 100).

`User Score` — оценка пользователей (от 0 до 10).

`Rating` — рейтинг организации ESRB (англ. Entertainment Software Rating Board). Эта ассоциация определяет рейтинг компьютерных игр и присваивает им подходящую возрастную категорию.
</div>

### Содержимое проекта

---
<br/>
<div class="alert alert-info">

1. Загрузка данных и знакомство с ними

2. Проверка ошибок в данных и их предобработка

3. Фильтрация данных

4. Категоризация данных

5. Итоговый вывод
</div>

## Загрузка данных и знакомство с ними

<br/>
<div class="alert alert-info">

- Загрузка необходимых библиотек Python и данных датасета `/datasets/new_games.csv`.
</div>

In [1]:
import pandas as pd
import numpy as np
import warnings

In [2]:
df = pd.read_csv('/datasets/new_games.csv')

<br/>
<div class="alert alert-info">

- Знакомство с данными: первые строки и результат метода `info()`.
</div>

In [3]:
df.head(10)

,Name,Platform,Year of Release,Genre,NA sales,EU sales,JP sales,Other sales,Critic Score,User Score,Rating
0,Wii Sports,Wii,2006.0,Sports,41.36,28.96,3.77,8.45,76.0,8,E
1,Super Mario Bros.,NES,1985.0,Platform,29.08,3.58,6.81,0.77,NaN,NaN,NaN
2,Mario Kart Wii,Wii,2008.0,Racing,15.68,12.76,3.79,3.29,82.0,8.3,E
3,Wii Sports Resort,Wii,2009.0,Sports,15.61,10.93,3.28,2.95,80.0,8,E
4,Pokemon Red/Pokemon Blue,GB,1996.0,Role-Playing,11.27,8.89,10.22,1.00,NaN,NaN,NaN
5,Tetris,GB,1989.0,Puzzle,23.20,2.26,4.22,0.58,NaN,NaN,NaN
6,New Super Mario Bros.,DS,2006.0,Platform,11.28,9.14,6.5,2.88,89.0,8.5,E
7,Wii Play,Wii,2006.0,Misc,13.96,9.18,2.93,2.84,58.0,6.6,E
8,New Super Mario Bros. Wii,Wii,2009.0,Platform,14.44,6.94,4.7,2.24,87.0,8.4,E
9,Duck Hunt,NES,1984.0,Shooter,26.93,0.63,0.28,0.47,NaN,NaN,NaN


In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16956 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16954 non-null  object 
 1   Platform         16956 non-null  object 
 2   Year of Release  16681 non-null  float64
 3   Genre            16954 non-null  object 
 4   NA sales         16956 non-null  float64
 5   EU sales         16956 non-null  object 
 6   JP sales         16956 non-null  object 
 7   Other sales      16956 non-null  float64
 8   Critic Score     8242 non-null   float64
 9   User Score       10152 non-null  object 
 10  Rating           10085 non-null  object 
dtypes: float64(4), object(7)
memory usage: 1.4+ MB


In [5]:
# создаём копию датасета для обработки, чтобы сохранить исходные данные
df_copy = df.copy()

<br/>
<div class="alert alert-info">

- Предоставленные данные соответствуют описанию. Датасет содержит 16956 строк и 11 столбцов. 

- В 6 столбцах есть пропуски данных, наибольшее количество пропусков наблюдается в столбце `Critic Score`, в котором содержится информаци об оценке игры критиками. Также по высокому количеству пропусков выделяются столбцы `User Score` и `Rating`, содержащие информацию об оценках пользователей и рейтинге игры.

- Следует отметить, что не все столбцы, содержащие числа имеют числовой тип данных. Столбцы `EU sales`, `JP sales` и `User Score` имеют тип данных `object`. 
</div>

---

##  Проверка ошибок в данных и их предобработка


### Названия, или метки, столбцов датафрейма

In [6]:
# названия всех столбцов
df_copy.columns

Index(['Name', 'Platform', 'Year of Release', 'Genre', 'NA sales', 'EU sales',
       'JP sales', 'Other sales', 'Critic Score', 'User Score', 'Rating'],
      dtype='object')

In [7]:
# приведение названий столбцов к стилю snake case
df_copy.columns = df.columns.str.lower().str.replace(' ', '_')

In [8]:
# проверка названий после изменения
df_copy.columns

Index(['name', 'platform', 'year_of_release', 'genre', 'na_sales', 'eu_sales',
       'jp_sales', 'other_sales', 'critic_score', 'user_score', 'rating'],
      dtype='object')

### Типы данных


<br/>
<div class="alert alert-info">

В столбцах по продажам `EU sales`, `JP sales` и `User Score` тип данных не числовой, хотя содержаться в них числа. Возможно, некоторые значения введены строкой, поэтому библиотека `pandas` не смогла сделать числовой тип данных для этих столбцов. Необходимо проверить значения в данных столбцах.
</div>

In [9]:
# проверка значений в столбце eu_sales
df_copy['eu_sales'].unique()

array(['28.96', '3.58', '12.76', '10.93', '8.89', '2.26', '9.14', '9.18',
       '6.94', '0.63', '10.95', '7.47', '6.18', '8.03', '4.89', '8.49',
       '9.09', '0.4', '3.75', '9.2', '4.46', '2.71', '3.44', '5.14',
       '5.49', '3.9', '5.35', '3.17', '5.09', '4.24', '5.04', '5.86',
       '3.68', '4.19', '5.73', '3.59', '4.51', '2.55', '4.02', '4.37',
       '6.31', '3.45', '2.81', '2.85', '3.49', '0.01', '3.35', '2.04',
       '3.07', '3.87', '3.0', '4.82', '3.64', '2.15', '3.69', '2.65',
       '2.56', '3.11', '3.14', '1.94', '1.95', '2.47', '2.28', '3.42',
       '3.63', '2.36', '1.71', '1.85', '2.79', '1.24', '6.12', '1.53',
       '3.47', '2.24', '5.01', '2.01', '1.72', '2.07', '6.42', '3.86',
       '0.45', '3.48', '1.89', '5.75', '2.17', '1.37', '2.35', '1.18',
       '2.11', '1.88', '2.83', '2.99', '2.89', '3.27', '2.22', '2.14',
       '1.45', '1.75', '1.04', '1.77', '3.02', '2.75', '2.16', '1.9',
       '2.59', '2.2', '4.3', '0.93', '2.53', '2.52', '1.79', '1.3', '2.6',
   

In [10]:
# проверка значений в столбце jp_sales
df_copy['jp_sales'].unique()

array(['3.77', '6.81', '3.79', '3.28', '10.22', '4.22', '6.5', '2.93',
       '4.7', '0.28', '1.93', '4.13', '7.2', '3.6', '0.24', '2.53',
       '0.98', '0.41', '3.54', '4.16', '6.04', '4.18', '3.84', '0.06',
       '0.47', '5.38', '5.32', '5.65', '1.87', '0.13', '3.12', '0.36',
       '0.11', '4.35', '0.65', '0.07', '0.08', '0.49', '0.3', '2.66',
       '2.69', '0.48', '0.38', '5.33', '1.91', '3.96', '3.1', '1.1',
       '1.2', '0.14', '2.54', '2.14', '0.81', '2.12', '0.44', '3.15',
       '1.25', '0.04', '0.0', '2.47', '2.23', '1.69', '0.01', '3.0',
       '0.02', '4.39', '1.98', '0.1', '3.81', '0.05', '2.49', '1.58',
       '3.14', '2.73', '0.66', '0.22', '3.63', '1.45', '1.31', '2.43',
       '0.7', '0.35', '1.4', '0.6', '2.26', '1.42', '1.28', '1.39',
       '0.87', '0.17', '0.94', '0.19', '0.21', '1.6', '0.16', '1.03',
       '0.25', '2.06', '1.49', '1.29', '0.09', '2.87', '0.03', '0.78',
       '0.83', '2.33', '2.02', '1.36', '1.81', '1.97', '0.91', '0.99',
       '0.95', '2.0'

In [11]:
# проверка значений в столбце user_score
df_copy['user_score'].unique()

array(['8', nan, '8.3', '8.5', '6.6', '8.4', '8.6', '7.7', '6.3', '7.4',
       '8.2', '9', '7.9', '8.1', '8.7', '7.1', '3.4', '5.3', '4.8', '3.2',
       '8.9', '6.4', '7.8', '7.5', '2.6', '7.2', '9.2', '7', '7.3', '4.3',
       '7.6', '5.7', '5', '9.1', '6.5', 'tbd', '8.8', '6.9', '9.4', '6.8',
       '6.1', '6.7', '5.4', '4', '4.9', '4.5', '9.3', '6.2', '4.2', '6',
       '3.7', '4.1', '5.8', '5.6', '5.5', '4.4', '4.6', '5.9', '3.9',
       '3.1', '2.9', '5.2', '3.3', '4.7', '5.1', '3.5', '2.5', '1.9', '3',
       '2.7', '2.2', '2', '9.5', '2.1', '3.6', '2.8', '1.8', '3.8', '0',
       '1.6', '9.6', '2.4', '1.7', '1.1', '0.3', '1.5', '0.7', '1.2',
       '2.3', '0.5', '1.3', '0.2', '0.6', '1.4', '0.9', '1', '9.7'],
      dtype=object)

<br/>
<div class="alert alert-info">

В столбцах `eu_sales`, `jp_sales` и `user_score` встречаются строковые значения 'unknown' и 'tbd', которые нужно заменить на пропуски.
</div>

In [12]:
# замена строковых значений 'unknown' и 'tbd' на пустую строку в столбцах eu_sales, jp_sales и user_score
df_copy['eu_sales'] = df_copy['eu_sales'].replace('unknown', np.nan)
df_copy['jp_sales'] = df_copy['jp_sales'].replace('unknown', np.nan)
df_copy['user_score'] = df_copy['user_score'].replace('tbd', np.nan)

In [13]:
# преобразование типов данных на числовые в обработанных столбцах
df_copy['eu_sales'] = df_copy['eu_sales'].astype('float64')
df_copy['jp_sales'] = df_copy['jp_sales'].astype('float64')
df_copy['user_score'] = df_copy['user_score'].astype('float64')

### Наличие пропусков в данных

<font color='#777778'>

- Посчитайте количество пропусков в каждом столбце в абсолютных и относительных значениях.


In [14]:
# количество пропусков по столбцам
df_copy.isna().sum()

name                  2
platform              0
year_of_release     275
genre                 2
na_sales              0
eu_sales              6
jp_sales              4
other_sales           0
critic_score       8714
user_score         9268
rating             6871
dtype: int64

In [15]:
# доля пропусков по столбцам
df_copy.isna().sum() / len(df)

name               0.000118
platform           0.000000
year_of_release    0.016218
genre              0.000118
na_sales           0.000000
eu_sales           0.000354
jp_sales           0.000236
other_sales        0.000000
critic_score       0.513918
user_score         0.546591
rating             0.405225
dtype: float64

In [16]:
# альтернативный способ посчёта пропусков
missing_table = pd.DataFrame({
    'Количество пропусков': df_copy.isnull().sum(),
    'Доля пропусков (%)': (df_copy.isnull().mean() * 100).round(2)
})

missing_table = missing_table.sort_values('Количество пропусков', ascending=False)
missing_table

,Количество пропусков,Доля пропусков (%)
user_score,9268,54.66
critic_score,8714,51.39
rating,6871,40.52
year_of_release,275,1.62
eu_sales,6,0.04
jp_sales,4,0.02
name,2,0.01
genre,2,0.01
platform,0,0.00
na_sales,0,0.00


<br/>
<div class="alert alert-info">

- Наибольшее количество пропусков наблюдается в столбцах с оценкой критиков, оценкой пользователей и рейтингом игры.Скорее всего, это значения не были проставлены, пользователи и критики не оценивали игру. Рейтинг ESRB не был определен, поскольку игра не прошла оценку в специализированной организации.

- Небольшое количество пустых значений наблюдается в столбцах с годом выпуска игры, жанром и с данными о продажах в Европе и Японии. Год выпуска мог быть пропущен из-за неясности точной даты выпуска игры, жанр может быть не присвоен игре, возможно из-за спорного оформления, когда нет явных признаков какого-то одного жанра. Данные по продажам могут отсутствовать из-за сложностей сбора такой информации в разных странах. 

- Данные с пропущенными значениями по году выпуска в анализ не попадут, поскольку для анализа будут отбраны данные за период с 2000 по 2013 год.

- У двух игр отсутствует название. 
</div>

<br/>
<div class="alert alert-info">

Заполнение и обработка пропусков

- В столбце `name` пропуски заменим на значение Unknown.

- В столбце `genre` пропуски заменим на значение Unknown.

- Пропуски в столбце с годом выпуска не будут обрабатываться, поскольку они пропадут после фильтрации данных до нужного среза.

- Пропуски в столбцах по продажам игр в Европе и Японии заменим на среднее значение по продажам в зависимости от платформы и года выхода игры.

- Пропуски в оценках и рейтинге останутся, поскольку такие данные сложно предположить.
</div>

In [17]:
# заполнение пропусков в столбце name
df_copy['name'] = df_copy['name'].fillna('Unknown')

In [18]:
# заполнение пропусков в столбце genre
df_copy['genre'] = df_copy['genre'].fillna('Unknown')

In [19]:
# заполнение пропусков в данных по количеству проданных игр в Европе и Японии
# функции для подбора среднего значения в зависимости от платформы и года выпуска

def eu_sales(row):
    if pd.isna(row['eu_sales']):
        eu_group = df_copy[(df_copy['platform'] == row['platform']) 
                           & (df_copy['year_of_release'] == row['year_of_release'])]
        return eu_group['eu_sales'].mean()
    else:
        return row['eu_sales']

def jp_sales(row):
    if pd.isna(row['jp_sales']):
        jp_group = df_copy[(df_copy['platform'] == row['platform']) 
                           & (df_copy['year_of_release'] == row['year_of_release'])]
        return jp_group['jp_sales'].mean()
    else:
        return row['jp_sales']

In [20]:
# применение функций для замены пропусков
df_copy['eu_sales'] = df_copy.apply(eu_sales, axis=1)
df_copy['jp_sales'] = df_copy.apply(jp_sales, axis=1)

### Явные и неявные дубликаты в данных

In [21]:
# проверка уникальных названий платформ
df_copy['platform'].unique()

array(['Wii', 'NES', 'GB', 'DS', 'X360', 'PS3', 'PS2', 'SNES', 'GBA',
       'PS4', '3DS', 'N64', 'PS', 'XB', 'PC', '2600', 'PSP', 'XOne',
       'WiiU', 'GC', 'GEN', 'DC', 'PSV', 'SAT', 'SCD', 'WS', 'NG', 'TG16',
       '3DO', 'GG', 'PCFX'], dtype=object)

In [22]:
# приведение названий платформ к верхнему регистру
df_copy['platform'] = df_copy['platform'].str.upper()

In [23]:
# проверка названий платформ после корректировки
df_copy['platform'].unique()

array(['WII', 'NES', 'GB', 'DS', 'X360', 'PS3', 'PS2', 'SNES', 'GBA',
       'PS4', '3DS', 'N64', 'PS', 'XB', 'PC', '2600', 'PSP', 'XONE',
       'WIIU', 'GC', 'GEN', 'DC', 'PSV', 'SAT', 'SCD', 'WS', 'NG', 'TG16',
       '3DO', 'GG', 'PCFX'], dtype=object)

In [24]:
# проверка названий жанров
df_copy['genre'].unique()

array(['Sports', 'Platform', 'Racing', 'Role-Playing', 'Puzzle', 'Misc',
       'Shooter', 'Simulation', 'Action', 'Fighting', 'Adventure',
       'Strategy', 'Unknown', 'MISC', 'ROLE-PLAYING', 'RACING', 'ACTION',
       'SHOOTER', 'FIGHTING', 'SPORTS', 'PLATFORM', 'ADVENTURE',
       'SIMULATION', 'PUZZLE', 'STRATEGY'], dtype=object)

<br/>
<div class="alert alert-info">

- В названиях жанров встречаются неявные дубликаты. В некоторых случаях одни и те же названия написаны в разных регистрах. Необходимо привести все названия жанров к одному регистру, чтобы сделать их одинаковыми.
</div>

In [25]:
# приведение названий жанров к нижнему регистру
df_copy['genre'] = df_copy['genre'].str.lower()

In [26]:
# проверка названий жанров после корректировки
df_copy['genre'].unique()

array(['sports', 'platform', 'racing', 'role-playing', 'puzzle', 'misc',
       'shooter', 'simulation', 'action', 'fighting', 'adventure',
       'strategy', 'unknown'], dtype=object)

In [27]:
# проверка написания годов
year_list = list(df_copy['year_of_release'].unique())
year_list_sorted = sorted(year_list)
year_list_sorted

[1980.0,
 1981.0,
 1982.0,
 1983.0,
 1984.0,
 1985.0,
 1986.0,
 1987.0,
 1988.0,
 1989.0,
 1990.0,
 1991.0,
 1992.0,
 1993.0,
 1994.0,
 1995.0,
 1996.0,
 1997.0,
 1998.0,
 1999.0,
 2000.0,
 2001.0,
 2002.0,
 2003.0,
 2004.0,
 2005.0,
 2006.0,
 2007.0,
 2008.0,
 2009.0,
 2010.0,
 2011.0,
 2012.0,
 2013.0,
 2014.0,
 2015.0,
 2016.0,
 nan]

<br/>
<div class="alert alert-info">

- В написании годов неявных дубликатов не наблюдается.
</div>

In [28]:
# проверка названий рейтинга
df_copy['rating'].unique()

array(['E', nan, 'M', 'T', 'E10+', 'K-A', 'AO', 'EC', 'RP'], dtype=object)

<br/>
<div class="alert alert-info">

В рейтингам по возрастным категориям есть следующие возможные

Возрастные категории ESRB:

EC (Early Childhood) — Для детей младшего возраста (3+).

E (Everyone) — Для всех возрастов (6+).

E10+ (Everyone 10+) — Для всех старше 10 лет.

T (Teen) — Для подростков от 13 лет.

M (Mature) — Для взрослой аудитории от 17 лет.

AO (Adults Only) — Только для взрослых, 18+.

RP (Rating Pending) — Рейтинг ожидается. 

K-A (Kids to Adults) – это старый рейтинг, который в настоящее время не используется, и был заменен на E (Everyone / Для всех), означавший, что игра подходит для всех возрастов. Возможно, после фильтрации данных по необходимому срезу лет строк с таким рейтингом не останется, но лучше привести ретинг к единообразию заранее, на случай если понадобится изучить другой временной срез данных.
</div>

In [29]:
# замена значения K-A на E в значениях рейтинга 
df_copy['rating'] = df_copy['rating'].replace('K-A', 'E')

In [30]:
# проверка значений рейтинга после замены
df_copy['rating'].unique()

array(['E', nan, 'M', 'T', 'E10+', 'AO', 'EC', 'RP'], dtype=object)

<br/>
<div class="alert alert-info">

Проверка и обработка явных дубликатов в данных
</div>

In [31]:
# считаем количество дубликатов
df_copy.duplicated().sum()

241

In [32]:
# выводим строки-дубликаты
duplicates = df_copy[df_copy.duplicated()]
duplicates

,name,platform,year_of_release,genre,na_sales,eu_sales,jp_sales,other_sales,critic_score,user_score,rating
268,Batman: Arkham Asylum,PS3,2009.0,action,2.24,1.31,0.07,0.61,91.0,8.9,T
368,James Bond 007: Agent Under Fire,PS2,2001.0,shooter,1.90,1.13,0.10,0.41,72.0,7.9,T
717,God of War: Ascension,PS3,2013.0,action,1.23,0.63,0.04,0.35,80.0,7.5,M
823,Wipeout: The Game,WII,2009.0,misc,1.94,0.00,0.00,0.12,NaN,NaN,NaN
848,Rayman Raving Rabbids: TV Party,WII,2008.0,misc,0.72,1.08,0.00,0.23,73.0,7.7,E10+
...,...,...,...,...,...,...,...,...,...,...,...
16671,Fullmetal Alchemist: Prince of the Dawn,WII,2009.0,adventure,0.00,0.00,0.01,0.00,NaN,NaN,NaN
16753,Routes PE,PS2,2007.0,adventure,0.00,0.00,0.01,0.00,NaN,NaN,NaN
16799,Transformers: Prime,WII,2012.0,action,0.00,0.01,0.00,0.00,NaN,NaN,NaN
16912,Metal Gear Solid V: The Definitive Experience,XONE,2016.0,action,0.01,0.00,0.00,0.00,NaN,NaN,M


<br/>
<div class="alert alert-info">

- В данных присутствуют явные дубликаты. 241 строку из 169566 нужно удвлить, чтобы убрать явные дубликаты. Убираем явные дубликаты с помощью метода `drop_duplicates()`. Сохраняем новый датафрейм без дубликатов, оставляем только первую строку из дублирующихся.
</div>

In [33]:
# сортируем датафрейм по всем столбцам
df_sorted = df_copy.sort_values(by=list(df_copy.columns))

# удаляем дубликаты
df_no_duplicates = df_sorted.drop_duplicates()

<br/>
<div class="alert alert-info">

- Часть строк в процессе предобработки была удалена. Чтобы вывести их количество в абсолютном и относительном значениях необходимо сравнить количество строк в датафрейме до обработки (`df`) и количество строк в датафрейме после обработки (`df_no_duplicates`)
</div>

In [37]:
# количество строк в исходном датафрейме
initial_row_count = df.shape[0]

# количество строк в датафрейме после обработки
final_row_count = df_no_duplicates.shape[0]

# cчитаем сколько строк было удалено
deleted_rows = initial_row_count - final_row_count

# cчитаем относительное количество удаленных строк
deleted_share = round(deleted_rows / initial_row_count, 3)

# выводим результаты
print(f"Количество строк в исходном датафрейме: {initial_row_count}")
print(f"Количество строк в датафрейме после обработки: {final_row_count}")
print(f"Количество удаленных строк: {deleted_rows}")
print(f"Доля удаленных строк: {deleted_share}")

Количество строк в исходном датафрейме: 16956
Количество строк в датафрейме после обработки: 16715
Количество удаленных строк: 241
Доля удаленных строк: 0.014


### Общий промежуточный вывод после предобработки данных

<br/>
<div class="alert alert-info">

- Для анализа был предоставлен датасет содержащий информацию по названиям игр, годам выпуска, платформам, жанрам, продажам в разных частях мира, оценкам и рейтингам. Названия столбцов были унифицированы, а именно, приведены к стилю написания `snake case`. В данных были обнаружены пропуски, неявные и явные дубликаты. Пропуски были заполнены там, где это было целесообразно, а именно в названиях игр и данным по продажам. Неявные дубликаты были обнарудены в названиях жанров игр. Строки с явными дубликатами были удалены. 
</div>

---

## Фильтрация данных


<br/>
<div class="alert alert-info">

- Делаем срез данных за период с 2000 по 2013 год с помощью метода `loc`.
</div>

In [34]:
df_actual = df_no_duplicates.loc[(df_no_duplicates['year_of_release'] >= 2000) 
                                 & (df_no_duplicates['year_of_release'] <= 2013)]
df_actual.head()

,name,platform,year_of_release,genre,na_sales,eu_sales,jp_sales,other_sales,critic_score,user_score,rating
3394,Frozen: Olaf's Quest,3DS,2013.0,platform,0.27,0.27,0.00,0.05,NaN,NaN,NaN
3906,Frozen: Olaf's Quest,DS,2013.0,platform,0.21,0.26,0.00,0.04,NaN,NaN,NaN
2478,Tales of Xillia 2,PS3,2012.0,role-playing,0.20,0.12,0.45,0.07,71.0,7.9,T
8460,.hack//G.U. Vol.1//Rebirth,PS2,2006.0,role-playing,0.00,0.00,0.17,0.00,NaN,NaN,NaN
7182,.hack//G.U. Vol.2//Reminisce,PS2,2006.0,role-playing,0.11,0.09,0.00,0.03,NaN,NaN,NaN


In [35]:
# поверка среза по годам после фильтрации
period_start = df_actual['year_of_release'].min()
period_finish = df_actual['year_of_release'].max()
print(f"Наименьший год в срезе данных: {period_start}")
print(f"Наибольший год в срезе данных: {period_finish}")

Наименьший год в срезе данных: 2000.0
Наибольший год в срезе данных: 2013.0


In [36]:
#проверяем значения в столбце rating
df_actual['rating'].unique()

array([nan, 'T', 'E', 'E10+', 'M', 'EC', 'AO', 'RP'], dtype=object)

<br/>
<div class="alert alert-info">

Рейтинг K-A исчез из данных, поскольку это устаревшее значение, и скорее всего, после 2000 года им перестали пользоваться.

Здесь уже можно не проверять названия рейтингов, поскольку они исправлены выше, но эта часть осталась, чтобы было понятно о чём комментарии.
</div>

---

## Категоризация данных


In [37]:
warnings.filterwarnings('ignore')

In [38]:
# разделение на категории по оценкам пользователей
df_actual['user_score_category'] = pd.cut(df_actual['user_score'], 
                                          bins=[0, 3, 8, 10], labels=["Low", "Medium", "High"], right=False)

In [39]:
# разделение на категории по оценкам критиков
df_actual['critic_score_category'] = pd.cut(df_actual['critic_score'], 
                                            bins=[0, 30, 80, 100], labels=["Low", "Medium", "High"], right=False)

In [40]:
# проверка получившегося итгового датафрейма
df_actual.head()

,name,platform,year_of_release,genre,na_sales,eu_sales,jp_sales,other_sales,critic_score,user_score,rating,user_score_category,critic_score_category
3394,Frozen: Olaf's Quest,3DS,2013.0,platform,0.27,0.27,0.00,0.05,NaN,NaN,NaN,NaN,NaN
3906,Frozen: Olaf's Quest,DS,2013.0,platform,0.21,0.26,0.00,0.04,NaN,NaN,NaN,NaN,NaN
2478,Tales of Xillia 2,PS3,2012.0,role-playing,0.20,0.12,0.45,0.07,71.0,7.9,T,Medium,Medium
8460,.hack//G.U. Vol.1//Rebirth,PS2,2006.0,role-playing,0.00,0.00,0.17,0.00,NaN,NaN,NaN,NaN,NaN
7182,.hack//G.U. Vol.2//Reminisce,PS2,2006.0,role-playing,0.11,0.09,0.00,0.03,NaN,NaN,NaN,NaN,NaN


In [41]:
# количество игр в каждой категории по оценкам пользователей
grouped_user_score = df_actual.groupby('user_score_category')['name'].count()
grouped_user_score 

user_score_category
Low        116
Medium    4081
High      2286
Name: name, dtype: int64

In [42]:
# количество игр в каждой категории по оценкам критиков
grouped_critic_score = df_actual.groupby('critic_score_category')['name'].count()
grouped_critic_score 

critic_score_category
Low         55
Medium    5422
High      1692
Name: name, dtype: int64

<br/>
<div class="alert alert-info">

- Топ-7 платформ по количеству выпущенных игр
</div>

In [43]:
grouped_platforms = df_actual.groupby('platform')['name'].count()
grouped_platforms.sort_values(ascending=False).head(7)

platform
PS2     2127
DS      2120
WII     1275
PSP     1180
X360    1121
PS3     1087
GBA      811
Name: name, dtype: int64

<br/>
<div class="alert alert-info">

- Продажи игр разных жанров
</div>

In [44]:
# группировка суммарных продаж по жанрам игр
sales_genres = df_actual.groupby('genre')['na_sales', 'eu_sales', 'jp_sales', 'other_sales'].sum().round(2)

<br/>
<div class="alert alert-info">

- Чтобы явно увидеть региональные предпочтения игроков проводим сортировку по продажам в Северной Америке и в Японии.
</div>

In [45]:
# сортировка по продажам в Северной Америке
sales_genres.sort_values(by = 'na_sales', ascending=False)

,na_sales,eu_sales,jp_sales,other_sales
genre,,,,
action,679.74,390.17,102.64,152.51
sports,543.26,293.85,72.62,111.71
shooter,416.21,228.50,18.41,78.64
misc,350.22,183.41,79.89,66.91
racing,263.57,177.89,25.98,66.10
platform,250.76,132.78,52.58,39.79
role-playing,248.83,127.52,210.55,42.88
simulation,159.54,97.40,37.85,27.49
fighting,148.34,67.82,38.53,29.00


In [46]:
# сортировка по продажам в Японии
sales_genres.sort_values(by = 'jp_sales', ascending=False)

,na_sales,eu_sales,jp_sales,other_sales
genre,,,,
role-playing,248.83,127.52,210.55,42.88
action,679.74,390.17,102.64,152.51
misc,350.22,183.41,79.89,66.91
sports,543.26,293.85,72.62,111.71
platform,250.76,132.78,52.58,39.79
fighting,148.34,67.82,38.53,29.00
simulation,159.54,97.40,37.85,27.49
adventure,76.49,43.01,33.74,12.95
strategy,45.09,30.76,26.76,8.50


<br/>
<div class="alert alert-info">

Интересно отметить, что предпочтения игроков в разных странах отличаются. В Японии наибольшей популярностью пользуются игры жанра `role-playing`, в то время как в Северной Америке и Европе лидирующую позицию занимают игры жанра `action`.
</div>

---

## Итоговый вывод


<br/>
<div class="alert alert-info">

- Проведен анализ данных по обзору компьютерных игр.

- Проект включал в себя предварительный просмотр данных и их корректировку. А именно проведен предпросмотр данных и проверка их на корректность. Унифицированы названия платформ и жанров игр. Обработаны пропуски в данных и дубликаты.

- Для анализа отобран срез данных за период с 2000 по 2013 годы.

- Проведена категоризация данных по оценкам пользователей и критиков с подсчетом количества игр, попавших в обозначенные категории. Как показал анализ наибольшее количество игр попадает в среднюю категорию по оценке. Критики более строго подходят к оценке игр, высокую оценку критиков получило меньшее количество игр по сравнению с высокой оценкой пользователей.

- По количеству выпущенных игр лидируют платформы `PS2` и `DS`. На остальных платформах выпущенно значительно меньшее количество игр.

- Популярность игр разных жанров в разных регионах отличается. Если сравнить продажи игр по жанрам в Америке - Европе и в Японии, то видим, что в Японии самам популярным жанром явлется `role-playing`. В то время как в Америке - Европе наиболее популярный жанр - `action`.
</div>